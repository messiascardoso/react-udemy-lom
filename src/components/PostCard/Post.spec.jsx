import { render, screen } from "@testing-library/react";
import { PostCard } from ".";

const props = {
  title: 'title1',
  body: 'body 1',
  id: 1,
  cover: 'img/img.png'
};

describe('<PostCard /> ', () => {
  it('should should render Postcard correctly', () => {
    render(<PostCard {...props} />);
    // const {debug} = render(<PostCard {...props} />);
    // debug();
    expect(screen.getByRole('img', {name: 'title1' })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', {name: 'title1 1' })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props}/>);
    expect(container.firstChild).toMatchSnapshot();
  });

});