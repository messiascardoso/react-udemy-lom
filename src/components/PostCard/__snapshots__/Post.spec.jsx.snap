// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard />  should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title1
       
      1
    </h1>
    <p>
      body 1
    </p>
  </div>
</div>
`;
